modules/@apostrophecms/piece-page-type/index.js

```diff
@@ -244,10 +244,16 @@ module.exports = {
           if (!req.aposParentPageCache) {
             req.aposParentPageCache = {};
           }
-          req.aposParentPageCache[pieceName] = pages;
+          if (pages?.length) {
+            req.aposParentPageCache[pieceName] = pages;
+          }
         }
         results.forEach(function (piece) {
-          const parentPage = self.chooseParentPage(req.aposParentPageCache[pieceName], piece);
+          const parentPage = self.chooseParentPage(
+            req.aposParentPageCache[pieceName] || [],
+            piece
+          );
+
```

Comment:
Real changes are here:
We don't set the cache if no data. This allow to retry to get the doc in draft after tryihng publish version but having no parent page (because this one is draft only).